### PR TITLE
Support TypeScript 4.1

### LINKED ISSUE #2383: Support TypeScript 4.1

### PR DESCRIPTION
**Pull request by saberduck**
Fixes #2383 

TS 4.1 requires NodeJS 10. We also update Dockerfile used for CI.

Some fixes required to pass ruling with new version:
- Fixed FPs by ignoring MemberAccess for S3758
- changed `globalReturn` to `false` to avoid additional scope between global and module scope
- changed highlighting of symbols to avoid runtime failure when typeannotation contains reference to the symbol

See also #2480 and #2481

### CODE REVIEW

File: eslint-bridge/src/rules/csrf.ts [hunk 1]
```diff
@@ -51,9 +51,9 @@ export const rule: Rule.RuleModule = {
     function checkIgnoredMethods(node: estree.Property) {
       if (node.value.type === 'ArrayExpression') {
         const arrayExpr = node.value;
-        const unsafeMethods = arrayExpr.elements.filter(
-          e => isLiteral(e) && typeof e.value === 'string' && !SAFE_METHODS.includes(e.value),
-        );
+        const unsafeMethods = arrayExpr.elements
+          .filter(isLiteral)
```
Comment (saberduck) [inline eslint-bridge/src/rules/csrf.ts:55]: required for `isLiteral` to be typeguard to have correct type inference.

**Review by andrey-tyukin-sonarsource — APPROVED** [review]
LGTM

Just a few tiny & purely optional formatting / refactoring suggestions here and there.

File: eslint-bridge/src/rules/declarations-in-global-scope.ts [hunk 2]
```diff
@@ -22,11 +22,15 @@ import { Rule } from 'eslint';
 
 export const rule: Rule.RuleModule = {
   create(context: Rule.RuleContext) {
+    function findModuleScope() {
```
[the remark this answers is not in the document]
Comment (saberduck) [reply eslint-bridge/src/rules/declarations-in-global-scope.ts:25]: I decided to refactor as you suggested

File: eslint-bridge/tests/analyzer.test.ts [hunk 3]
```diff
@@ -166,8 +166,8 @@ describe('#analyzeJavaScript', () => {
       fileContent: codeToTest,
       tsConfigs: [tsConfig],
     });
-    expect(parsingError).toBeDefined();
-    expect(parsingError.message).toContain('Debug Failure');
+    expect(parsingError).toBeUndefined();
+    // expect(parsingError.message).toContain('Debug Failure');
```
[the remark this answers is not in the document]
Comment (saberduck) [reply eslint-bridge/tests/analyzer.test.ts:170]: indeed, I added the comment that this test is no longer testing what's supposed to test

**Comment by MarkLyck** [comment]
@saberduck do you. have any idea when this will make it's way into SonarCloud?

I just wasted 2 days trying to figure out why SonarCloud wasn't analyzing my code, and finally found it's because it's running on TS 3.x.

Happy to see this is already merged! 

Was wondering if I could just wait for this to be deployed on SonarCloud or if I need to do extra work arounds to get it working by next week?

**Pull request merged**